backend/routes_chemicals.py

```
@@ -20,148 +31,176 @@ def decorated_function(*args, **kwargs):
 def register_chemical_routes(app):
     # Get all chemicals
     @app.route('/api/chemicals', methods=['GET'])
+    @handle_errors
     def chemicals_route():
```

Pylint (3.3.7)

[refactor] Too many branches (16/12)

(R0912)